### PR TITLE
chore: validate content URL

### LINKED ISSUE #665: [l10n] CHS: "Sharing URL" string is unlocalized in Options
**Description**
"Sharing URL" string is unlocalized in Options > Add via sharing URL in zh_CN locale

**To Reproduce**
1. Install extension
2. Right click on extension button > Options

**Expected behavior**
String is localized:
config_shareurl_placeholder#message - Sharing URL: https://www.hlx.live/sidekick... - 共享 URL: https://www.hlx.live/sidekick...

**Screenshots**
chs
fra

**Additional context**
String is localized in Aladdin
aladdin

### LINKED ISSUE #688: Content URL field can be empty in Options > Edit
**Description**
Required Content URL field can be cleared and left empty in Options > Edit project

**To Reproduce**
1. Go to project folder
2. Click on extension button > "Add this project"
3. Right click on extension button -> "Options"
4. Click "Edit" button in "Projects"
5. Clear "Content URL" field
6. Click "Save"

**Expected behavior**
Error shows when Content URL field is clear or field is optional

**Screenshots**

### LINKED ISSUE #698: move sharepoint edit info call from service worker to tab
SP calls from the service worker tend to receive 401 even with credentials included, whereas SP calls from the tab usually succeed.

### LINKED ISSUE #700: Unlocalised string "Invalid hostname" in Options > Edit
**Description**
String "Invalid hostname" is unlocalized

**To Reproduce**
1. Install AEM sidekick extension
2. Go to project folder in Google Drive
3. Click on extension button > "Add this project"
4. Right click on extension button -> "Options"
5. Click "Edit" button in "Projects"
6. Enter "1" character in "Production Hostname (optional)"

**Expected behavior**
String is localized

**Additional context**
String is localized in GLaaS
String ID: config_invalid_host#message

**Screenshots**
6 41 1 fra
6 41 1 jpn

### PR DESCRIPTION
**Pull request by asthabh23**
Fix #665
Fix #688 
Fix #698
Fix #700

### CODE REVIEW

File: src/extension/options.js [hunk 1]
```diff
@@ -36,6 +36,10 @@ function isValidGitHubURL(giturl) {
     && Object.keys(getGitHubSettings(giturl)).length === 3;
 }
 
+function isInvalidMountpoint(mountpoints) {
```
Comment (rofe) [inline src/extension/options.js:39]: I think I would turn this into `isValidMountpoint` (to align with `isValidHost` etc)

**Review by rofe — CHANGES_REQUESTED** [review]

File: src/extension/_locales/en/messages.json [hunk 2]
```diff
@@ -173,6 +173,9 @@
   "config_invalid_host": {
     "message": "Invalid hostname"
   },
+  "config_invalid_mountpoints": {
+    "message": "Invalid Content URL"
```
Comment (rofe) [inline src/extension/_locales/en/messages.json:177]: @rgeraghty can you add the translations for this new english item?

Comment (rgeraghty) [reply src/extension/_locales/en/messages.json:177]: @rofe hope to have translation tomorrow,

**Pull request merged**